### PR TITLE
TEST-#3254: add check asv benchmarks on pure pandas

### LINKED ISSUE #3254: Add check ASV benchmarks on pure Pandas
A run of benchmarks on the Pandas was added, the results in the dashboard can be viewed using `machine=xeon-e5-pandas` filter. This check is needed to find errors at an earlier stage (Modin CI).

Example:
http://modin.org/modin-bench/#benchmarks.TimeAppend.time_append?machine=xeon-e5-pandas&python=3.8&p-shapes=((500000%2C%2020)%2C%20(1000000%2C%2010))

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3254 
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -390,15 +390,19 @@ jobs:
           git fetch upstream
           if git diff upstream/master --name-only | grep -q "^asv_bench/"; then
               asv machine --yes
-              asv run --quick --show-stderr --launch-method=spawn -b ^benchmarks -b ^io -b ^scalability \
-                | sed "/failed$/ s/^/##[error]/" | tee benchmarks.log
-              # check omnisci backend
-              MODIN_BACKEND=omnisci MODIN_EXPERIMENTAL=true asv run --quick -b ^omnisci \
-                --show-stderr --launch-method=forkserver --config asv.conf.omnisci.json \
-                | sed "/failed$/ s/^/##[error]/" | tee benchmarks.log
-              if grep "failed" benchmarks.log > /dev/null ; then
```
Comment (vnlitvinov) [inline .github/workflows/ci.yml:399]: why `sed` and `grep` are no longer there? were they not needed?

Comment (anmyachev) [reply .github/workflows/ci.yml:399]: Previously, ASV would always return 0, even if the benchmarks failed. In this case, we used parsing of logs in order to return an error code other than 0. The new version of the program has `--strict` option that returns an error code other than 0 if benchmarks fail, which eliminates the need to parse the logs.

Comment (vnlitvinov) [reply .github/workflows/ci.yml:399]: Where is ASV version update then?..

Comment (anmyachev) [reply .github/workflows/ci.yml:399]: This feature already in ASV that we take from master https://github.com/modin-project/modin/blob/7727c23adc58c470b6008dc38519da7d21c3ac0b/environment-dev.yml#L42

**Review by vnlitvinov — APPROVED** [review]
LGTM

**Pull request merged**